crates/polars-ops/src/frame/pivot/mod.rs

```diff
@@ -192,112 +192,125 @@ fn pivot_impl(
 
     let mut final_cols = vec![];
 
+    let column_column_name = &columns[0];
+    let mut binding = pivot_df.clone();
+    let pivot_df: &DataFrame = if columns.len() > 1 {
+        let new_cols = columns
+            .iter()
+            .map(|s| pivot_df.column(s).unwrap().clone())
+            .collect::<Vec<_>>();
+        let new_col = StructChunked::new(&columns[0], &new_cols)
+            .unwrap()
+            .into_series();
+        binding.with_column(new_col).unwrap()
+    } else {
+        pivot_df
+    };
```

Comment:
in the base case, `pivot_df` isn't modified

does the `clone` here just increase the reference count, and so is zero-cost here as in that case `pivot_df` isn't modified?
if so, then the base case (single-column) is unaffected by these changes